File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int n = tokens.length;
        for(int i = 0; i < n; i++) {
            String token = tokens[i];
            switch(token) {
                case "+" : {
                    int a = stack.pop();
                    int b = stack.pop();
                    stack.push(a + b);
                    break;
                }
                case "-" : {
                    int a = stack.pop();
                    int b = stack.pop();
                    stack.push(b - a);
                    break;
                }

                case "/" : {
                    int a = stack.pop();
                    int b = stack.pop();
                    stack.push(b / a);
                    break;
                }
                case "*" : {
                    int a = stack.pop();
                    int b = stack.pop();
                    stack.push(b * a);
                    break;
                }
                default: {
                    stack.push(Integer.parseInt(token));
                }
            }
        }
        return stack.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack = [];
    let operators = ["+", "-","*","/"];
    for(let token of tokens) {
        if(operators.indexOf(token) >= 0) {
            let op2 = stack.pop();
            let op1 =  stack.pop();
            let ans;
            if(token == '+') {
                ans = op1 + op2;
            } else if(token == '-') {
                ans = op1 - op2;
            } else if(token == '*') {
                ans = op1 * op2;
            } else if(token == '/') {
                ans = parseInt(op1 / op2);
            }
            stack.push(ans);
        } else {
            stack.push(parseInt(token));
        }
    }
    return stack.pop();
};